### PR TITLE
[backport] snap: add GH actions jobs to release the snap package

### LINKED ISSUE #3098: snap: add GH actions jobs to release the snap package
backport https://github.com/kata-containers/kata-containers/pull/1007

### PR DESCRIPTION
**Pull request by devimc**
Use Github actions to build and release the snap package automatically
when a new tag is pushed.

fixes #3098

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

**Review by wainersm — APPROVED** [review]

File: .github/workflows/snap-release.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Install Snapcraft
+        uses: samuelmeuli/action-snapcraft@v1
+        with:
+          snapcraft_token: ${{ secrets.snapcraft_token }}
+
+      - name: Build snap
+        run: |
+          sudo apt-get install -y git git-extras
+          kata_url="https://github.com/kata-containers/runtime"
+          latest_version=$(git ls-remote --tags ${kata_url}  | egrep -o "refs.*" | egrep -v "\-alpha|\-rc|{}" | egrep -o "[[:digit:]]+\.[[:digit:]]+\.[[:digit:]]+" | sort -V -r | head -1)
```
Comment (jodh-intel) [inline .github/workflows/snap-release.yaml:22]: OCD nit: double space before first pipe :smile:

File: .github/workflows/snap-release.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+          # Check if the current tag is the latest tag
+          if echo -e "$latest_version\n$current_version" | sort -C -V; then
+            # Current version is the latest version, build it
+            snapcraft -d snap --destructive-mode
+          fi
+
+      - name: Upload snap
+        run: |
+          snap_version="$(echo ${GITHUB_REF} | cut -d/ -f3)"
+          snap_file="kata-containers_${snap_version}_amd64.snap"
```
Comment (jodh-intel) [inline .github/workflows/snap-release.yaml:33]: Might be worth creating a variable for the architecture. OOI, what would need to happen to build this for the other arches?

**Review by jodh-intel — APPROVED** [review]

File: .github/workflows/snap-release.yaml (hunk 2 above)
Comment (jodh-intel) [reply .github/workflows/snap-release.yaml:33]: Ah - I hadn't noticed the backport label so feel free to ignore this comment (first sentence atleast ;)

Comment (devimc) [reply .github/workflows/snap-release.yaml:33]: afaik github actions run on x86 only :neckbeard:

**Pull request merged**